scipy/stats/tests/test_distributions.py

```diff
@@ -371,6 +371,195 @@ def test_pdf_zero(self):
         assert_equal(stats.geninvgauss.pdf(2e6, 50, 2), 0)
 
 
+class TestGenHyperbolic(object):
+    def setup_method(self):
+        np.random.seed(1234)
+
+    def test_pdf_r(self):
+        # test against R package GeneralizedHyperbolic
+        # x <- seq(-10, 10, length.out = 10)
+        # GeneralizedHyperbolic::dghyp(
+        #    x = x, lambda = 2, alpha = 2, beta = 1, delta = 1.5, mu = 0.5
+        # )
+        vals_R = np.array([
+            2.94895678275316e-13, 1.75746848647696e-10, 9.48149804073045e-08,
+            4.17862521692026e-05, 0.0103947630463822, 0.240864958986839,
+            0.162833527161649, 0.0374609592899472, 0.00634894847327781,
+            0.000941920705790324
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        x = np.linspace(-10, 10, 10)
+        gh = stats.genhyperbolic(
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
+            )
+
+        assert_allclose(gh.pdf(x), vals_R, atol=1e-9)
```

Comment:
atol=1e-9 does not work for the first test value with is of order 1e-13. if atol=0, what is the best rtol we can use?

Reply:
Probably one can go down a couple of o.o.m, but 1e-9 looks fine enough for me.

Reply:
Alright,  i have reduced tolerance at the smallest o.o.m possible here: https://github.com/scipy/scipy/pull/13298/commits/e0c794de681862b95ffd798f92dc4ec5dd22d8ad. Given the current implementation, i do not think it is possible to go below that

Reply:
I did not expain it very well: the check is `abs(actual - desired) <= atol + rtol * abs(desired)`

if `desired = 1e-13` and `actual = 1e-20`, the test would pass if `atol = 1e-9` although the relative error is huge. so one should rather put atol = 0. since the cdf values are <= 1, it is enough to specify rtol, e.g. the default is 1e-7. (if you compare huge numbers, this might not be a good idea, e.g. if desired = 1e9, rtol=1e-7 would allow an absolute error of 100)

Reply:
Thanks for the clarification. I was not aware of the behaviour of `assert_allclose`. Will check with `atol=0` and see.

Reply:
Looks good now. Most of the time, rtol is < 1e-10, I guess CDF is challenging, 1e-6 seems reasonable.